### PR TITLE
:rotating_light: :rotating_light: :sparkles:  Source Jira: Migrate `IssueComments` and `IssueWorklogs` streams to low code

### PR DESCRIPTION
**Pull request by tolik0**
## What
Migrate `IssueComments` and `IssueWorklogs` streams to low code based on the CDK update to add an incremental dependency to the parent stream. Corresponding CDK PR: https://github.com/airbytehq/airbyte/pull/38211

## How
The `IssueComments` and `IssueWorklogs` streams have been migrated to low code. Their parent stream, Issues, will now be read incrementally using the state from the child streams. The parent state will be saved to the child state.

## User Impact
This change is not reversible.
This PR introduces a breaking change due to the state format modification. Migration to the new version will be done automatically, but the new state format is incompatible with the previous version. The first sync after migration may be slow as the connector needs to reread all parent records. 

**Old state format:**

```
"stream_state": {
  "updated": "2222-05-08T03:04:45.056-0700"
}
```
**New state format:**

```
"stream_state": {
  "state": [
    {
        "updated": "2222-04-15T11:40:03.372-0700"
    }
  ],
  "parent_state": {
    "issues": {
      "states": [
        {
          "partition": {
            "parent_slice": {},
            "project_id": "10000"
          },
          "cursor": {
            "updated": "2222-10-12T13:43:50.735-0700"
          }
        },
        {
          "partition": {
            "parent_slice": {},
            "project_id": "10016"
          },
          "cursor": {
            "updated": "2222-07-05T12:57:51.258-0700"
          }
        }
      ]
    }
  }
}
```

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [x] NO ❌

### CODE REVIEW

File: docs/integrations/sources/jira.md [hunk 1]
```diff
@@ -127,6 +127,7 @@ The Jira connector should not run into Jira API limitations under normal usage.
 
 | Version | Date       | Pull Request                                               | Subject                                                                                                                                                          |
 |:--------|:-----------|:-----------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 3.1.0   | 2024-08-16 | [38612](https://github.com/airbytehq/airbyte/pull/38612) | Migrate IssueComments and IssueWorklogs streams to low code                                                                                                      |
```
Comment (girarda) [inline docs/integrations/sources/jira.md:130]: let's be super clear that this is not revertible

**Comment by tolik0** [comment]
Regression tests - https://github.com/airbytehq/airbyte/actions/runs/11554415703/job/32157573020
The value of the cursor changed due to new format:
Previous - `2021-04-15T11:39:46.574000-07:00`, New - `2021-04-15T11:39:46.574-07:00`

File: docs/integrations/sources/jira.md (hunk 1 above)
Comment (tolik0) [reply docs/integrations/sources/jira.md]: Updated the message

**Review by maxi297 — APPROVED** [review]
LGTM! I'm just wondering why we emit so many more state messages in the new version based on the regression tests [here](https://github.com/airbytehq/airbyte/pull/38612#issuecomment-2441585896)

**Comment by tolik0** [comment]
The difference in state count is due to [the Python implementation](https://github.com/airbytehq/airbyte/blob/86718a18cc31124948058a12a9fc57eae1f5db16/airbyte-integrations/connectors/source-jira/source_jira/streams.py#L93) previously emitting state messages at intervals, whereas now we emit them after every slice.

**Review by darynaishchenko — APPROVED** [review]

**Pull request merged**